bolt-sidecar/src/driver.rs

```diff
@@ -354,3 +390,29 @@ impl<C: StateFetcher, ECDSA: SignerECDSA> SidecarDriver<C, ECDSA> {
         }
     }
 }
+
+/// Pick a pubkey to sign constraints with.
+///
+/// Rationale:
+/// - If there are no delegatee keys, try to use the validator key directly if available.
+/// - If there are delegatee keys, try to use the first one that is available in the list.
+fn pick_public_key(
+    validator: BlsPublicKey,
+    available: Vec<BlsPublicKey>,
+    delegatees: Vec<BlsPublicKey>,
```

Comment:
Any chance we can make these `HashSet<BlsPublicKey>` compared to `Vec`? So we don't need to loop over every time, and this should also address your concerns about the logic being a bit inefficient.

Reply:
good point